app/services/globus_digest_generator.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+# GlobusDigestGenerator
+#
+# Add digests to Globus DROs if the files currently lack them. Existing files
+# will not be overwritten.
+class GlobusDigestGenerator
+  # @param [Cocina::Model] cocina a DRO to add digests to
+  # @param [Hash] globus_ids a mapping of filenames to their location on disk
+  # @param [Boolean] validate whether to build the new DRO using validation
```

Comment:
Based on how I'm reading, the `validate` flag controls whether we're creating a `DRO` or a `RequestDRO` (basically the same thing except the latter lacks a druid/external ID). If that's correct, we could consider renaming this argument to something like `request`? Alternatively (again, if this is correct), we could also do away with the flag entirely and let the digest generator sniff the given Cocina for presence/absence of a druid/external ID and build the appropriate object based on that?

Reply:
@mjgiarlo yes your interpretation was correct. I thought about having it sniff the Cocina to see if a request was needed, but then I double guessed myself and thought it seemed a bit too magical. But if you had the same thought I'll go with that.

Reply:
I tried using `cocina.externalIdentifier` to sniff it but it threw a missing method error when I used it on a RequestDRO. I decided instead I tested to see what type of Cocina it is working with.